CMakeLists.txt

```diff
@@ -1,12 +1,12 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.16)
 project(rapidcheck CXX)
 
 # Don't warn about symbol visibility for static libraries with CMake 3.3 and later.
 if(POLICY CMP0063)
   cmake_policy(SET CMP0063 NEW)
 endif()
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
This is the only line that is different from upstream in our fork. The rest of the changes are merging upstream to our branch and removing in-house implementations for C++11 features that were missing from old Android NDKs years ago.